third_party/cpgfunctionEP-0.3.2/CMakeLists.txt

```
@@ -0,0 +1,86 @@
+cmake_minimum_required(VERSION 3.6)
+project(cpgfunctionEP)
+
+set(CMAKE_CXX_STANDARD 17)
+if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --std=c++17")
+else()
+    set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+    add_compile_definitions(_USE_MATH_DEFINES)
+endif()
+
+# Everything will need to access this
+include_directories(include)
+# Third party includes
+include_directories(third_party/eigen-3.3.9)  # Eigen for LU decomposition
+include_directories(third_party/qdt-master)
+
+# Create a DLL with an API
+add_library(cpgfunctionEP SHARED
+        src/boreholes.cpp
+        src/gfunction.cpp
+        src/heat_transfer.cpp
+        src/interpolation.cpp
+        src/utilities.cpp
+        src/coordinates.cpp
+        src/statistics.cpp
+        src/segments.cpp
+        src/blas.cpp)
+
+# It depends on threads
+find_package(Threads REQUIRED)
+target_link_libraries(cpgfunctionEP ${CMAKE_THREAD_LIBS_INIT})
```

Modernize the target_link_libraries by explcitly passing the type (PUBLIC, PRIVATE, INTERFACE). Here it's just passing `-lthreads` basically, so it's an INTERFACE.

```suggestion
target_link_libraries(cpgfunctionEP INTERFACE Threads::Threads)
```